iothub/device/src/Transport/AmqpIoT/AmqpIoTSession.cs

```diff
@@ -255,16 +281,38 @@ TimeSpan timeout
             amqpLinkSettings.AddProperty(AmqpIoTErrorAdapter.TimeoutName, timeout.TotalMilliseconds);
             amqpLinkSettings.AddProperty(AmqpIoTErrorAdapter.ClientVersion, deviceIdentity.ProductInfo.ToString());
             amqpLinkSettings.AddProperty(AmqpIoTErrorAdapter.ApiVersion, ClientApiVersionHelper.ApiVersionString);
-            if (CorrelationId != null)
+            if (correlationId != null)
             {
-                amqpLinkSettings.AddProperty(AmqpIoTErrorAdapter.ChannelCorrelationId, CorrelationId);
+                amqpLinkSettings.AddProperty(AmqpIoTErrorAdapter.ChannelCorrelationId, correlationId);
             }
 
-            ReceivingAmqpLink receivingLink = new ReceivingAmqpLink(amqpLinkSettings);
-            receivingLink.AttachTo(amqpSession);
-            await receivingLink.OpenAsync(timeout).ConfigureAwait(false);
-            if (Logging.IsEnabled) Logging.Exit(typeof(AmqpIoTSession), deviceIdentity, $"{nameof(OpenReceivingAmqpLinkAsync)}");
-            return new AmqpIoTReceivingLink(receivingLink);
+            try
+            {
+                ReceivingAmqpLink receivingLink = new ReceivingAmqpLink(amqpLinkSettings);
+                receivingLink.AttachTo(amqpSession);
+                await receivingLink.OpenAsync(timeout).ConfigureAwait(false);
+                return new AmqpIoTReceivingLink(receivingLink);
+            }
+            catch (Exception e) when (!e.IsFatal())
+            {
+                Exception ex = AmqpIoTExceptionAdapter.ConvertToIoTHubException(e, amqpSession);
```

Comment:
Please encapsulate - same as above.

Reply:
Please propose the right way of doing it. Table

Reply:
See above for ideas.